src/cellular/dataCollection.cpp

```diff
@@ -0,0 +1,123 @@
+#include "ensembleTypes.hpp"
+#include "product.hpp"
+#include "system.hpp"
+#include "cli/flog.hpp"
+#include "cli/conio.hpp"
+#include "util.hpp"
+#include "imu/imu.hpp"
+
+#include "Particle.h"
+
+typedef struct Ensemble10_eventData_
+{
+    double temperature;
+    int32_t water;
+    int32_t acc[3];
+    int32_t ang[3];
+    int32_t mag[3];
+    int32_t location[2];
+    uint8_t hasGPS;
+    uint32_t accumulateCount;
+}Ensemble10_eventData_t;
+
+static Ensemble10_eventData_t ensemble10Data;
+
+
+void SS_ensemble10Func()
+{
+    int32_t lat, lng;
+    float temp;
+    uint8_t water;
+    float accelData[3] = {0,0,0};
+    float gyroData[3] = {0,0,0};
+    float magData[3] = {0,0,0};
+
+    LocationPoint point;
+    LocationService& instance= LocationService::instance();
+
+    
+    bool hasGPS = false;
+    Ensemble10_eventData_t* pData = &ensemble10Data;
+
+    #pragma pack(push, 1)
+    struct
+    {
+        EnsembleHeader_t header;
+        union{
+            Ensemble10_data_t ens10;
+            Ensemble11_data_t ens11;
+        }data;
+    }ensData;
+    #pragma pack(pop)
+
+    getAccelerometer(accelData, accelData + 1, accelData + 2);
+    getGyroscope(gyroData, gyroData + 1, gyroData + 2);
+    getMagnetometer(magData, magData + 1, magData + 2);
+
+    /**
+     * Check pre-processer defines to switch between 
+     * ICM and MAX temp sensor
+     */
+    #if USE_ICM_TEMP_SENSOR
+        getTemperature(&temp);
+    #else 
+        temp = pSystemDesc->pTempSensor->getTemp();
+    #endif
+        
+    instance.getLocation(point);
+
+    if(point.locked == 1 && point.satsInView > 4)
+    {
+        hasGPS = true;
+        lat = point.latitude;
+        lng = point.longitude;
+    }
+    else
+    {
+        hasGPS = false;
+        lat = pData->location[0];
+        lng = pData->location[1];
+    }
+
+    water = 0; // TODO - waiting on water sensor integration
+
+    // Accumulate measurements
+    pData->temperature += temp;
+    pData->water += water;
+    pData->acc[0] += B_TO_N_ENDIAN_2(accelData[0]);
+    pData->acc[1] += B_TO_N_ENDIAN_2(accelData[1]);
+    pData->acc[2] += B_TO_N_ENDIAN_2(accelData[2]);
+    pData->ang[0] += B_TO_N_ENDIAN_2(gyroData[0]);
+    pData->ang[1] += B_TO_N_ENDIAN_2(gyroData[1]);
+    pData->ang[2] += B_TO_N_ENDIAN_2(gyroData[2]);
+    pData->mag[0] += B_TO_N_ENDIAN_2(magData[0]);
+    pData->mag[1] += B_TO_N_ENDIAN_2(magData[1]);
+    pData->mag[2] += B_TO_N_ENDIAN_2(magData[2]);
```

Comment:
Why are we accumulating after endianness change?  You must not do native math after endianness change, since the math is no longer valid.